design/pool-configuration.md

```diff
@@ -0,0 +1,359 @@
+# Address Pool configuration options
+
+## Summary
+
+Following @johananl's initiative in [this issue](https://github.com/metallb/metallb/issues/630)
+this enhancement tries to collect and classify various configuration enhancement requests
+requested over time.
+
+## Motivation
+
+The way advanced users consume MetalLB is evolving and exposed a need for more complex configuration.
+For example, the need for isolation and multi-tenancy induced requests for being able to advertise some specific address pools from a subset of the nodes, or
+to allocate them only to a specific namespace, to mention some.
+
+### Goals
+
+* Collect and classify all the requests coming from different issues, in order to have a better understanding
+  and an organic implementation, avoiding focus on and implementation of only a specific feature, risking causing conflicts afterwards.
+* Provide a list of the possible configuration enhancements and highlight dependencies and conflicts among them (if any).
+* Highlight the cases where the solution must be BGP or L2 specific
+
+### Non-Goals
+
+* Covering only a specific use case.
+* Providing a L2 only or BGP only solution where the problem might be expressed in a more generic way.
+
+## Proposal
+
+For all the IPs belonging to an address pool, we need to address three main configuration degrees:
+
+* Announcing the IP from a subset of nodes
+* Announcing the IP to a subset of BGP neighbors
+* Using a given AddressPool only for a subset of services
```

Comment:
Actually we seem to have **two** different concerns here:

1. Ensuring service X receives an IP from address pool Y.
2. Ensuring no other service besides X receives an IP from address pool Y.

The 1st concern is independent from the 2nd: I may want to ensure service X gets its IP from pool Y but I don't care if other services also get their IPs from pool Y.

Reply:
Makes sense, I think this will be covered by your rephrasing above. It doesn't add much value to split the item into two.